### ISSUE TITLE
Restrict a good to user-groups

### ISSUE THREAD

**Issue by udkde** [wielebenwir/commons-booking-2#27]
It can be necessary for some legal reason that a good is not open to everybody. For example when the service is payed for by the members of a cooperative and you have to be member of that cooperative.

Can you add a function that restricts access to a good by email address?

**Comment by flegfleg** [comment]
Hi, there are a lot of [plugins for Wordpress](https://wordpress.org/plugins/search/user+registration+activate/) that allow you to restrict user access, if you just need to restrict site access to users "activated" by the admin. 

> by email address

You mean by domain? E.g. everybody with an email address ending in `@kasimir-lastenrad.de` gets automatically activated? That should be possible with a plugin, too. 

Or you use a invitation code (this is possible with the [WP Members](https://wordpress.org/plugins/wp-members/) plugin, i implemented that for [another project](https://www.fearexchange.com/register/): If you have an access code, you can enter right away, otherwise you have to be manually approved by an admin.

**Comment by udkde** [comment]
Exactly, filter by email domain. But not for the whole sharing system but only for some goods that are restricted.

**Comment by flegfleg** [comment]
Ok. Access groups itself should be handled by another plugin (assign role based on the condition you described), since there are a lot of plugins for that already. 

we‘ll look into how to assign Booking/viewing rights for individual items based on roles. 

Once CB2.0 is released, I think this could probably be achieved in a hack-a-thon over a weekend.
